optuna/_callbacks.py

```diff
@@ -124,15 +119,33 @@ def __call__(self, study: "optuna.study.Study", trial: FrozenTrial) -> None:
     @staticmethod
     @experimental("2.8.0")
     def retried_trial_number(trial: FrozenTrial) -> Optional[int]:
-        """Return the number of the trial being retried.
+        """Return the number of the original trial being retried.
 
         Args:
             trial:
                 The trial object.
 
         Returns:
-            The number of the first failed trial. If not retry of a previous trial,
+            The number of the latest failed trial. If not retry of a previous trial,
             returns :obj:`None`.
         """
 
         return trial.system_attrs.get("failed_trial", None)
+
+    @staticmethod
+    @experimental("2.11.0")
+    def retry_history(trial: FrozenTrial) -> List[int]:
+        """Return the list of trial IDs from the original trial to the specified one.
```

Comment:
How about the following? I think it's easy to understand.
```suggestion
        """Return the list of retried trial IDs with respect to the specified trial.
```